src/main/java/cz/cvut/kbss/jsonld/annotation/JsonLdType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * JB4JSON-LD
+ * Copyright (C) 2025 Czech Technical University in Prague
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3.0 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.
+ */
+package cz.cvut.kbss.jsonld.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies that the class is an entity class and maps to an ontological class (RDFS or OWL).
+ * <p>
+ * This annotation is applied to the entity class.
+ * Do not provide iri on an abstract class.
```

Comment:
I would turn this condition around - if the IRI is empty, the class must be abstract. But the user may declare a class abstract and still provide type IRI for it (just like with `@OWLClass`).